### PR TITLE
Feature/audio_streams

### PR DESCRIPTION
**Pull request by hadjilucasL**
Hi !

Thank you very much for creating this ! 

I recently started playing around with GStreamer as I needed a python interface to get data from streaming audio sources. This library has been very helpful. I branched out and extended a bit the code to add support for audio capabilities. I thought you might like to merge it in. Let me know if you have any comments !

Kind regards,
Lucas

### CODE REVIEW

File: tests/test_pipeline.py [hunk 1]
```diff
@@ -72,22 +79,26 @@ def test_appsrc_no_caps():
...
         "appsink emit-signals=true"
     )
     with GstPipeline(cmd) as pipeline:
-        pipeline.set_appsrc_caps(width=320, height=240, framerate=10, format="GRAY8")
+        pipeline.set_appsrc_video_caps(
+            width=320, height=240, framerate=10, format="GRAY8"
+        )
         while pipeline:
-            pipeline.push(rand_array(channels=3))
+            pipeline.push(rand_array(channels=1))
```
Comment (hadjilucasL) [inline tests/test_pipeline.py:101]: Wasn't sure if this was an error or this is how the test was designed - The format is meant to be GRAY8 which single channel but the array being pushed has three channels.

File: tests/test_pipeline.py [hunk 2]
```diff
@@ -29,18 +27,29 @@ def test_appsink_buffers():
...
+    with GstPipeline(cmd) as pipeline:
+        while pipeline:
+            buffer = pipeline.pop()
+            if buffer:
+                count += 1
+                assert isinstance(buffer.data, np.ndarray)
+    assert count == num_buffers
+
+
+def test_tee(tmp_path):
```
Comment (hadjilucasL) [inline tests/test_pipeline.py:42]: uses the tmp_path fixture from pytest

File: src/gstreasy/wrapped_caps.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        16: np.dtype(np.int16),
+    }
+    format_info = GstAudio.AudioFormat.get_info(fmt)
+    return dtypes.get(format_info.depth, np.dtype(np.uint8))
+
+
+def _get_video_np_dtype(fmt: GstVideo.VideoFormat) -> np.dtype:
+    dtypes = {
+        8: np.dtype(np.uint8),
+        16: np.dtype(np.uint16),
```
Comment (hadjilucasL) [inline src/gstreasy/wrapped_caps.py:104]: I 'think' this is correct - i.e. pixels are all unsigned ints.

File: src/gstreasy/wrapped_caps.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    @property
+    def shape(self):
+        """Return shape of np.ndarray required to hold buffer with these Caps."""
+        return [self.height, self.width, self.channels]
+
+
+def _get_audio_np_dtype(fmt: GstAudio.AudioFormat) -> np.dtype:
+    dtypes = {
+        8: np.dtype(np.int8),
+        16: np.dtype(np.int16),
```
Comment (hadjilucasL) [inline src/gstreasy/wrapped_caps.py:95]: Need to double check this.

File: src/gstreasy/utils.py [hunk 5]
```diff
@@ -108,53 +77,6 @@ def make_caps(
     return caps
 
 
-def has_flag(value: GstVideo.VideoFormatFlags, flag: GstVideo.VideoFormatFlags) -> bool:
```
Comment (hadjilucasL) [inline src/gstreasy/utils.py:111]: Moved this under wrapped_caps.py

File: pyproject.toml [hunk 6]
```diff
@@ -34,7 +34,8 @@ dev = [
     "flake8-docstrings",
     "mypy",
     "pytest",
-    "tox"
+    "tox",
+    "build",
```
Comment (hadjilucasL) [inline pyproject.toml:38]: I was missing this in my local dev env.

**Comment by danofsteel32** [comment]
This all looks good to me. I've only ever needed to use gstreamer for video didn't even think about handling audio.

Thank you for taking the time/effort to make your changes available it's really appreciated!

**Pull request merged**